version: "3.9"

services:
  app:
    image: bytecom/4us.dev
    environment:
      - TZ=America/Fortaleza
      - NEXT_PUBLIC_APP_ENV=production
    deploy:
      replicas: 1
      update_config:
        parallelism: 1
        order: start-first
        failure_action: rollback
        delay: 5s
      rollback_config:
        parallelism: 0
        order: stop-first
      restart_policy:
        condition: any
        delay: 5s
        window: 120s
      labels:
        - "traefik.enable=true"
        - "traefik.http.routers.app.rule=Host(`app.bytecom.psi.br`)"
        - "traefik.http.routers.app.entrypoints=websecure"
        - "traefik.http.routers.app.tls.certresolver=myresolver"
        # - "traefik.http.routers.app-secure.rule=Host(`app.bytecom.psi.br`)"
        # - "traefik.http.routers.app-secure.entrypoints=websecure"
        # - "traefik.http.routers.app-secure.tls=true"
        # - "traefik.http.services.app.loadbalancer.server.port=3000"
        # - "traefik.http.services.app.loadbalancer.healthcheck.path=/"
        # - "traefik.http.services.app.loadbalancer.healthcheck.interval=10s"
        # - "traefik.http.services.app.loadbalancer.healthcheck.timeout=5s"
    networks:
      - net_app

networks:
  net_app:
    external: true
